### PR TITLE
[BACKLOG-32232] Fix for displaying urls correctly in File Open/Save D…

### PR DESCRIPTION
**Pull request by bmorrise**
…ialog

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
core,plugins/connections/ui,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **4718**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7054/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
core,plugins/connections/ui,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **4718**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7054/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 31s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
core,plugins/connections/ui,plugins/file-open-save-new/core
```
---
:ok_hand: All tests passed!
 
Tests run: **4718**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7054/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/model/VFSDirectory.java [hunk 1]
```diff
@@ -96,4 +99,26 @@ public static VFSDirectory create( String parent, FileObject fileObject, String
...
+    if ( obj == this ) {
+      return true;
+    }
+
+    if ( !( obj instanceof VFSFile ) ) {
+      return false;
+    }
+
+    VFSDirectory compare = (VFSDirectory) obj;
+    return compare.getProvider().equals( getProvider() )
```
Comment (NJtwentyone) [inline plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/model/VFSDirectory.java]: can't you use this for cleaner code:
`Objects.equals(compare.getProvider(), getProvider()) && Objects.equals(compare.getConnection(), getConnection()) && ...`

**Pull request merged**